dmoj/utils/aux_files.py

```diff
@@ -0,0 +1,61 @@
+import os
+import tempfile
+import traceback
+
+from dmoj.error import InternalError
+from dmoj.executors import executors
+from dmoj.executors.compiled_executor import CompiledExecutor
+
+
+def mktemp(data):
+    tmp = tempfile.NamedTemporaryFile()
+    tmp.write(data)
+    tmp.flush()
+    return tmp
+
+
+def compile_aux_files(filenames, lang=None, compiler_time_limit=None):
+    sources = {}
+
+    try:
+        for filename in filenames:
+            with open(filename, 'rb') as f:
+                sources[os.path.basename(filename)] = f.read()
+    except Exception:
+        traceback.print_exc()
+        raise IOError('Could not read source!')

```

Comment:
you probably want to just... let this exception happen so it gets collected by the IE handler